File: infrastructure/src/main/java/com/study/platform/domain/team/infrastructure/TeamMemberQueryRepositoryAdapter.java
package com.study.platform.domain.team.infrastructure;

import com.study.platform.domain.team.dto.response.TeamMemberResponse;
import com.study.platform.domain.team.model.TeamMemberRole;
import com.study.platform.domain.team.port.TeamMemberQueryPort;
import com.study.platform.global.jooq.JooqUtils;
import lombok.RequiredArgsConstructor;
import org.jooq.DSLContext;
import org.jooq.impl.DSL;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
@RequiredArgsConstructor
public class TeamMemberQueryRepositoryAdapter implements TeamMemberQueryPort {

    private final DSLContext dsl;

    @Override
    public List<TeamMemberResponse> findAllByTeamId(UUID teamId) {
        return dsl
                .select(
                        DSL.field("BIN_TO_UUID(m.id)").as("id"),
                        DSL.field("BIN_TO_UUID(u.id)").as("userId"),
                        DSL.field("u.nickname").as("nickname"),
                        DSL.field("m.role").as("role")
                )
                .from(DSL.table("team_members").as("m"))
                .join(DSL.table("users").as("u"))
                .on(DSL.field("m.user_id").eq(DSL.field("u.id")))
                .where(JooqUtils.uuidEq("m.team_id", teamId))
                .fetch(r -> new TeamMemberResponse(
                        UUID.fromString(r.get("id", String.class)),
                        UUID.fromString(r.get("userId", String.class)),
                        r.get("nickname", String.class),
                        TeamMemberRole.valueOf(r.get("role", String.class))
                ));
    }
}

File: infrastructure/src/main/kotlin/com/study/platform/domain/team/infrastructure/TeamMemberQueryRepositoryAdapter.kt
package com.study.platform.domain.team.infrastructure

import com.study.platform.domain.team.dto.response.TeamMemberResponse
import com.study.platform.domain.team.model.TeamMemberRole
import com.study.platform.domain.team.port.TeamMemberQueryPort
import com.study.platform.global.jooq.JooqUtils.binToUuid
import com.study.platform.global.jooq.JooqUtils.uuidEq
import com.study.platform.jooq.tables.references.TEAM_MEMBERS
import com.study.platform.jooq.tables.references.USERS
import org.jooq.DSLContext
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
class TeamMemberQueryRepositoryAdapter(
    private val dsl: DSLContext
) : TeamMemberQueryPort {

    override fun findAllByTeamId(teamId: UUID): List<TeamMemberResponse> {
        val m = TEAM_MEMBERS.`as`("m")
        val u = USERS.`as`("u")

        return dsl.select(
            binToUuid(m.ID).`as`("id"),
            binToUuid(u.ID).`as`("userId"),
            u.NICKNAME.`as`("nickname"),
            m.ROLE.`as`("role")
        )
            .from(m)
            .join(u).on(m.USER_ID.eq(u.ID))
            .where(uuidEq(m.TEAM_ID, teamId))
            .fetch { r ->
                TeamMemberResponse(
                    UUID.fromString(r.get("id", String::class.java)),
                    UUID.fromString(r.get("userId", String::class.java)),
                    r.get("nickname", String::class.java),
                    TeamMemberRole.valueOf(r.get("role", String::class.java))
                )
            }
    }
}
